From b74b2bd3a81c37a7a6d7dd90fff667419a58bc37 Mon Sep 17 00:00:00 2001
From: yifengling0 <yifengling0@users.noreply.github.com>
Date: Sat, 18 Jul 2026 00:22:43 +0800
Subject: [PATCH 051/104] fix(opengl32): preserve parsed legacy extension
 indices

---
 dlls/opengl32/unix_wgl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/opengl32/unix_wgl.c b/dlls/opengl32/unix_wgl.c
index fbf5654aeaac..9e0e4c9c6b0b 100644
--- a/dlls/opengl32/unix_wgl.c
+++ b/dlls/opengl32/unix_wgl.c
@@ -1100,7 +1100,7 @@ static void make_context_current( TEB *teb, const struct opengl_funcs *funcs, HD
     {
         enum opengl_extension extensions[GL_EXTENSION_COUNT];
         size_t extension_count = parse_extensions( (const char *)funcs->p_glGetString( GL_EXTENSIONS ), extensions);
-        for (i = 0; i < extension_count; i++) client->extensions[i] = TRUE;
+        for (i = 0; i < extension_count; i++) client->extensions[extensions[i]] = TRUE;
     }
 
     pthread_once( &once, init_enabled_extensions );

