From 74a94164286074a77cd3e3936c5dd8ccd66aa3c3 Mon Sep 17 00:00:00 2001
From: yifengling0 <yifengling0@users.noreply.github.com>
Date: Thu, 6 Aug 2026 05:30:57 +0800
Subject: [PATCH 097/104] fix(loader): keep mixed overlay declarations portable

---
 dlls/ntdll/loader.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/dlls/ntdll/loader.c b/dlls/ntdll/loader.c
index 904f8fcd25e3..301524a7a834 100644
--- a/dlls/ntdll/loader.c
+++ b/dlls/ntdll/loader.c
@@ -3314,12 +3314,14 @@ static NTSTATUS search_winehua_dxvk_overlay( LPCWSTR libname, UNICODE_STRING *nt
     UNICODE_STRING backend, root;
     NTSTATUS status = STATUS_DLL_NOT_FOUND;
     unsigned int i;
+    BOOL is_d3d12, is_mixed_vkd3d;
+    const WCHAR *root_name;
 
-    const BOOL is_d3d12 = !wcscmp( libname, d3d12W );
+    is_d3d12 = !wcscmp( libname, d3d12W );
     if (wcscmp( libname, d3d11W ) && wcscmp( libname, dxgiW ) && !is_d3d12)
         return status;
     if (get_env_var( backend_nameW, 0, &backend )) return status;
-    const BOOL is_mixed_vkd3d = !wcscmp( backend.Buffer, L"vkd3d_limited_500k" );
+    is_mixed_vkd3d = !wcscmp( backend.Buffer, L"vkd3d_limited_500k" );
     if (backend.Length < 5 * sizeof(WCHAR) ||
         (!is_mixed_vkd3d && wcsncmp( backend.Buffer, L"dxvk_", 5 )) ||
         (is_d3d12 && !is_mixed_vkd3d)) {
@@ -3327,7 +3329,7 @@ static NTSTATUS search_winehua_dxvk_overlay( LPCWSTR libname, UNICODE_STRING *nt
         return status;
     }
     RtlFreeUnicodeString( &backend );
-    const WCHAR *root_name = is_d3d12 ? vkd3d_root_nameW : dxvk_root_nameW;
+    root_name = is_d3d12 ? vkd3d_root_nameW : dxvk_root_nameW;
     if (get_env_var( root_name, 0, &root )) return status;
 
     for (i = 0; i < ARRAY_SIZE(archW); ++i)

